src/routes/exercise-finished/ExerciseFinishedScreen.tsx

```diff
@@ -56,66 +67,103 @@ interface Props {
 const ExerciseFinishedScreen = ({ navigation, route }: Props): ReactElement => {
   const { exercise, results, disciplineTitle, disciplineId, documents, closeExerciseAction, unlockedNextExercise } =
     route.params
-  const [message, setMessage] = React.useState<string>('')
+  const correctResults = results.filter(doc => doc.result === 'correct')
+  const percentageOfCorrectResults = correctResults.length / results.length
 
-  React.useEffect(() => {
-    const correctResults = results.filter(doc => doc.result === 'correct')
-    const correct = correctResults.length / results.length
-    if (unlockedNextExercise) {
-      setMessage(labels.results.unlockedExercise)
-    } else if (correct > 2 / 3) {
-      setMessage(labels.results.feedbackGood)
-    } else if (correct > 1 / 3) {
-      setMessage(labels.results.feedbackMedium)
-    } else {
-      setMessage(labels.results.feedbackBad)
-    }
-  }, [results])
+  const [message, setMessage] = useState<string>('')
+  const [resultColor, setResultColor] = useState<Color>(theme.colors.background)
+  const [buttonText, setButtonText] = useState<string>('')
+  const [navigationAction, setNavigationAction] = useState<() => void>(() => undefined)
+  const [icon, setIcon] = useState<ReactElement>()
 
-  const repeatExercise = (): void => {
+  const repeatExercise = (): void =>
     navigation.navigate(EXERCISES[exercise].screen, {
       documents,
       disciplineId,
       disciplineTitle,
       closeExerciseAction
     })
+
+  const startNextExercise = (): void => {
+    if (exercise + 1 < EXERCISES.length) {
+      navigation.navigate(EXERCISES[exercise + 1].screen, {
+        documents,
+        disciplineId,
+        disciplineTitle,
+        closeExerciseAction
+      })
+    }
   }
 
-  const checkResults = (): void => {
-    navigation.navigate('Result', route.params)
+  const navigateToNextModule = (): void => {
+    navigation.pop(2)
```

Comment:
this navigates me to the home screen when i finished the last exercise of a module. Why not directly navigate to this screen but pop back?

Reply:
It is not trivial to navigate to the next module or even the parent discipline selection. This solution is kind of a quick fix. Have a look at https://issues.tuerantuer.org/browse/LUN-364